cmd/kubeadm/app/cmd/reset.go

```diff
@@ -206,9 +237,17 @@ func newCmdReset(in io.Reader, out io.Writer, resetOptions *resetOptions) *cobra
 		if cmd.Flags().Lookup(options.NodeCRISocket) == nil {
 			// avoid CRI detection
 			// assume that the command execution does not depend on CRISocket when --cri-socket flag is not set
-			resetOptions.criSocketPath = kubeadmconstants.UnknownCRISocket
+			resetOptions.externalcfg.CRISocket = kubeadmconstants.UnknownCRISocket
 		}
-		return newResetData(cmd, resetOptions, in, out)
+		data, err := newResetData(cmd, resetOptions, in, out, false)
```

Comment:
Should we expose allowExperimental as a parameter instead of fixing it to false?

Reply:
I am not against the flag, but given we are trying to void too many flags, let's open to others for feedback. @neolit123 @pacoxu

Reply:
I think the flag is needed. Or the `kubeadm reset` cannot support v1beta4 ResetConfiguration.

/hold

Reply:
```
[root@daocloud ~]# ./kubeadm reset --config xx --allow-experimental-api
unknown flag: --allow-experimental-api
To see the stack trace of this error execute with --v=5 or higher
[root@daocloud ~]#  ./kubeadm reset --allow-experimental-api  --config xx
unknown flag: --allow-experimental-api
To see the stack trace of this error execute with --v=5 or higher
[root@daocloud ~]# cat xx | head -n 10
apiVersion: kubeadm.k8s.io/v1beta4
kind: ResetConfiguration
force: true
cleanupTmpDir: true
criSocket: unix:///var/run/containerd/containerd.sock
certificatesDir: /etc/kubernetes/pki
ignorePreflightErrors:
- a
- b
---

[root@daocloud ~]#  ./kubeadm reset --allow-experimental-api  --config xx
unknown flag: --allow-experimental-api
To see the stack trace of this error execute with --v=5 or higher

[root@daocloud ~]# ./kubeadm reset --config xx
experimental API spec: "kubeadm.k8s.io/v1beta4" is not allowed. You can use the --allow-experimental-api flag if the command supports it.
To see the stack trace of this error execute with --v=5 or higher
```

Reply:
we can just force this code path to allowExperimental = true? we shouldn't add the flag IMO.

Reply:
I am not sure if the flag `allow-experimental-api` is temporarily supported,  or will still available after the release of v1beta4,  anyway, I will add this flag to make this work.

Reply:
we shouldn't add the flag. it should be hardcoded to true.

Reply:
+1 to hardcode to true. Or this will be a fake support for Reset😄.

Reply:
mark this as resolved as the value is hardcoded to true now.

Reply:
/unhold